adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -288,78 +288,75 @@ private void handleEndpointConnectionWithAuthentication(final MqttEndpoint endpo
 
     void handleCredentialsResult(final AsyncResult<String> attempt, final MqttEndpoint endpoint, final UsernamePasswordCredentials authObject) {
         if (attempt.succeeded()) {
-            String logicalDeviceId = attempt.result();
-            LOG.debug("successfully authenticated device id [{}]", logicalDeviceId);
+            final String deviceId = attempt.result();
+            LOG.debug("successfully authenticated device id [{}]", deviceId);
             endpoint.accept(false);
 
             endpoint.publishHandler(message -> {
                 final ResourceIdentifier resource = ResourceIdentifier.fromString(message.topicName());
 
-                if (!validateCredentialsWithTopicStructure(resource, authObject.getTenantId(), logicalDeviceId)) {
+                // check that the device publishes to its device-id
+                if (!validateCredentialsWithTopicStructure(resource, authObject.getTenantId(), deviceId)) {
+
                     // if MQTT client does not conform to the topic structure, close the connection (MQTT has no way for errors)
                     endpoint.close();
                 } else {
-                    publishMessage(endpoint, authObject.getTenantId(), logicalDeviceId, message, resource);
+                    // the payload is ALWAYS published to the deviceId downstream
+                    publishMessage(endpoint, authObject.getTenantId(), deviceId, message, resource);
                 }
             });
 
             endpoint.closeHandler(v -> {
-                LOG.debug("connection closed with client [{}], authId [{}], deviceId [{}]",
-                        endpoint.clientIdentifier(), authObject.getAuthId(), logicalDeviceId);
+                LOG.debug("connection closed with tenantId [{}], authId [{}], deviceId [{}]",
```

Comment:
maybe something like "connection closed with client (tenantId [{}], authId [{}], deviceId [{}])" ?

Reply:
`closed connection to device [tenant-id: {}, auth-id: {}, device-id: {}]`